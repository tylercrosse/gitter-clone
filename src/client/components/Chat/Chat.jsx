import React       from 'react';
import { Link }    from 'react-router';
import { connect } from 'react-redux';
import { fetchMessages, addMessage } from '../../actions/';
import ChatMenu    from './ChatMenu.jsx';
import ChatHeader  from './ChatHeader.jsx';
import ChatToolbar from './ChatToolbar.jsx';
import ChatContent from './ChatContent.jsx';
import ChatInput   from './ChatInput.jsx';
import                  './chat.scss';

export class Chat extends React.Component {
  componentDidMount() {
    this.props.fetchMessages();
  }
  render() {
    const form = this.props.user.loggedIn ? (
      <ChatInput
        user={this.props.user}
        onMessageSubmit={this.props.addMessage}
      />
    ) : (
      <div className="chat-input">
        <Link to="/">Sign In</Link>
      </div>
    );
    return (
      <div className="chat-container">
        <ChatMenu />
        <main className="chat-header-wrapper">
          <ChatHeader user={this.props.user} />
          <div className="chat-and-toolbar-wrapper">
            <ChatToolbar />
            <div className="chat-wrapper">
              <ChatContent messages={this.props.messages} />
              {form}
            </div>
          </div>
        </main>
      </div>
    );
  }
}


export const mapStateToProps = (state) => ({
  messages: state.messages,
  user: state.user
});

export default connect(
  mapStateToProps,
  { fetchMessages, addMessage }
)(Chat);
