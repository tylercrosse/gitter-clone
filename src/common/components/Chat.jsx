import React from 'react';
import PropTypes from 'prop-types';

const Chat = ({ menu, main }) => (
  <div className="chat-container">
    {menu}
    {main}
  </div>
);

Chat.propTypes = {
  menu: PropTypes.object.isRequired,
  main: PropTypes.object.isRequired,
};

export default Chat;
