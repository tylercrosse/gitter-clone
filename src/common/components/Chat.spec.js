import React        from 'react';
import { shallow }  from 'enzyme';
import toJson       from 'enzyme-to-json';
import Chat         from './Chat';

const setup = (propOverrides) => {
  const props = Object.assign({
    main: <div>main</div>,
    menu: <div>menu</div>
  }, propOverrides);

  const component = <Chat {...props} />;

  return {
    props,
    component
  };
};

describe('<ChatMain />', () => {
  it('should render correctly', () => {
    const { component } = setup();
    const wrapper = shallow(component);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
