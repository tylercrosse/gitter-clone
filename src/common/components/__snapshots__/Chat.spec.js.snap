// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMain /> should render correctly 1`] = `
<div
  className="chat-container"
>
  <div>
    menu
  </div>
  <div>
    main
  </div>
</div>
`;
